theano/tensor/basic.py

```diff
@@ -4708,29 +4709,33 @@ def vertical_stack(*args):
     return concatenate(args, axis=0)
 
 
+
 class Reshape(Op):
     """Perform a reshape operation of the input x to the new shape shp.
-
     The number of dimensions to which to reshape to (ndim) must be
     known at graph build time.
-
     """
     view_map = {0: [0]}  # output 0 is potentially aliased to inputs [0]
     _f16_ok = True
 
     check_input = False
     __props__ = ("ndim",)
+    params_type = ParamsType(ndim=Scalar('int32'))
     # name does not participate because it doesn't affect computations
 
     def __init__(self, ndim, name=None):
-        self.ndim = ndim
+        self.ndim = int(ndim)
         if ndim < 0:
             raise ValueError("The output dimensions after reshape must be 0 or greater")
         assert name is None, 'name attribute for Reshape has been deprecated'
 
     def __str__(self):
         return '%s{%s}' % (self.__class__.__name__, self.ndim)
 
+
+    def get_params(self, node):
```

Comment:
When we use `ParamsType`, and if all params defined into `ParamsType` have corresponding op attributes, then we don't need to define `get_params()`: params values will be retrieved automatically.

Here, we define `ParamsType` with a param `ndim`, and this Op has an attribute named `ndim`, so we don't need to define `get_params()`.